crates/fuel-core/src/database/vm_database.rs

```diff
@@ -155,6 +155,7 @@ impl InterpreterStorage for VmDatabase {
             self.database
                 .get_block_id(block_height.into())?
                 .ok_or(not_found!("BlockId"))
+                .map(Into::into)
```

Comment:
Maybe we should consider promoting the `BlockId` type up to `fuel-types` and use that in the InterpreterStorage interface instead of a generic `Bytes32`?

Reply:
#896